hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java

```diff
@@ -614,6 +615,12 @@ private void loadOMHAConfigs(Configuration conf) {
             " system with " + OZONE_OM_SERVICE_IDS_KEY + " and " +
             OZONE_OM_ADDRESS_KEY;
         throw new OzoneIllegalArgumentException(msg);
+      } else if (!isOMAddressSet && found == 0) {
```

Comment:
We cannot add this condition here, because think of case
OZONE_OM_SERVICE_IDS_KEY = ns1,ns2
Because after one iteration of nameserviceID if we are not able to find, we should not say illegal configuration. We should try out all name services and see if it is matching with any nameservice. 

And also, we can eliminate iteration of for loop with OmUtils.emptyAsSingletonNull(omServiceIds))  and similar for nameServiceIds. As for HA this is a must required configuration which needs to be set.

Reply:
And also try out test cases with multiple name services set also. And for this test case, you can use ozone.om.node.id if needed.

Reply:
@bharatviswa504 Thanks for the review. You are right. Now I understand that it makes sense to walk through all configured service ids before declaring failure.

I should probably just put `found == 0` check outside the `serviceId` loop.